src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs

```diff
@@ -0,0 +1,226 @@
+                var solutionWithNewDocument = projectToBeUpdated.Solution.AddDocument(newDocumentId, newDocumentName, text: string.Empty);
+
+                // update the text for the new document
+                solutionWithNewDocument = solutionWithNewDocument.WithDocumentSyntaxRoot(newDocumentId, modifiedRoot, PreservationMode.PreserveIdentity);
+
+                // get the updated document, perform clean up like remove unused usings.
+                var newDocument = solutionWithNewDocument.GetDocument(newDocumentId);
+                newDocument = await CleanUpDocumentAsync(newDocument, cancellationToken).ConfigureAwait(false);
+
+                return newDocument.Project.Solution;
+            }
+
+            /// <summary>
+            /// update the original document and remove the type that was moved.
+            /// perform other fix ups as necessary.
+            /// </summary>
+            /// <param name="sourceDocument">original document</param>
+            /// <param name="typeNode">type that was moved to new document</param>
+            /// <param name="cancellationToken">a cancellation token</param>
+            /// <returns>an updated solution with the original document fixed up as appropriate.</returns>
+            private async Task<Solution> RemoveTypeFromSourceDocumentAsync(
+                Document sourceDocument, TTypeDeclarationSyntax typeNode, CancellationToken cancellationToken)
+            {
+                var documentEditor = await DocumentEditor.CreateAsync(sourceDocument, cancellationToken).ConfigureAwait(false);
+
+                AddPartialModifiersToTypeChain(documentEditor, typeNode);
+                documentEditor.RemoveNode(typeNode, SyntaxRemoveOptions.KeepNoTrivia);
+
+                var updatedDocument = documentEditor.GetChangedDocument();
+
+                updatedDocument = await CleanUpDocumentAsync(updatedDocument, cancellationToken).ConfigureAwait(false);
+
+                return updatedDocument.Project.Solution;
+            }
+
+            /// <summary>
+            /// Traverses the syntax tree of the forked document and
+            /// collects a list of nodes that are not being moved.
+            /// This list of nodes are then removed from the forked copy.
+            /// </summary>
+            /// <param name="root">root, of the syntax tree of forked document</param>
+            /// <param name="typeNode">node being moved to new document</param>
+            /// <returns>list of syntax nodes, to be removed from the forked copy.</returns>
+            private static IEnumerable<SyntaxNode> GetMembersToRemove(
+                SyntaxNode root, TTypeDeclarationSyntax typeNode)
+            {
+                HashSet<SyntaxNode> spine = new HashSet<SyntaxNode>();
+
+                // collect the parent chain of declarations to keep.
+                SyntaxNode node = typeNode;
+                while (node.Parent != null && !(node.Parent is TCompilationUnitSyntax))
```

Comment:
why do you skip the compilation unit? it's one of the nodes you would never remove.